internal/controller/children.go

```diff
@@ -101,14 +102,34 @@ func (c ChildResource[T]) Reconcile(ctx context.Context, clnt client.Client, cr
 			if !ok {
 				return fmt.Errorf("Could not cast when reconciling")
 			}
+			events := v1.EventList{}
```

Comment:
issue: This whole change should be moved to the `State` method on `ChildResourceUpdates` rather than here. The statusCallback function is meant only for cases where the status depends on the previous and current state - like hibernation. You do not need to know the previous state for this case so you do not need to do this here.

You can get the k8s client from `r *AmaltheaSessionReconciler` in `Status()` and then feed it through to `State()` which is called from within `Status()`.